flusher.go

```diff
@@ -166,6 +166,32 @@ func (s *Server) Flush(interval time.Duration, metricLimit int) {
 	s.logger.WithField("metrics", len(finalMetrics)).Info("Completed flush to Datadog")
 }
 
+func finalizeMetrics(hostname string, tags []string, finalMetrics []DDMetric) []DDMetric {
+	for i := range finalMetrics {
+		// Let's look for "magic tags" that override metric fields host and device.
+		for j, tag := range finalMetrics[i].Tags {
+			// This overrides hostname
+			if strings.HasPrefix(tag, "host:") {
+				// delete the tag from the list
+				finalMetrics[i].Tags = append(finalMetrics[i].Tags[:j], finalMetrics[i].Tags[j+1:]...)
+				// Override the hostname with the tag, trimming off the prefix
+				finalMetrics[i].Hostname = string(tag[5:len(tag)])
```

Comment:
`len(tag)` is superfluous, if the right parameter of the slice expression is empty then it goes to the end by default. same with the other one below